cloud-optimized-netcdf4-hdf5/index.qmd

```diff
@@ -1,19 +1,263 @@
+
+# How tos
+
+The examples below require the HDF5 library package is installed on your system. These commands will also work for NetCDF-4. While you can check for chunk size and shape with h5py, h5py is a high-level interface primarily for accessing datasets, attributes, and other basic HDF5 functionalities. h5py does not expose lower-level file options directly.
+
+## Commands in brief:
+
+* [`h5stat`](https://support.hdfgroup.org/documentation/hdf5/latest/_h5_t_o_o_l__s_t__u_g.html) prints stats from an existing HDF5 file.
+* [`h5repack`](https://support.hdfgroup.org/documentation/hdf5/latest/_h5_t_o_o_l__r_p__u_g.html) writes a new file with a new layout.
+* [`h5dump`](https://support.hdfgroup.org/documentation/hdf5/latest/_h5_t_o_o_l__d_p__u_g.html) displays objects from an HDF5 file.
+
+
+## How to check for consolidated metadata
+
+To be considered cloud-optimized, HDF5 files should be written with the `PAGE` file space management strategy (see also [File Space Management Strategies](https://support.hdfgroup.org/documentation/hdf5-docs/advanced_topics/FileSpaceManagement.html#strategies)). When using this strategy, HDF5 will write aggregate metadata and raw data into fixed-size pages [@jelenak2023].
+
+You can check the file space management strategy with the command line h5stat tool:
+
+``` bash
+h5stat -S infile.h5
+```
+
+This returns output such as:
+
+``` text
+Filename: infile.h5
+File space management strategy: H5F_FSPACE_STRATEGY_FSM_AGGR
+File space page size: 4096 bytes
+Summary of file space information:
+  File metadata: 2157376 bytes
+  Raw data: 37784376 bytes
+  Amount/Percent of tracked free space: 0 bytes/0.0%
+  Unaccounted space: 802424 bytes
+Total space: 40744176 bytes
+```
+
+Notice the strategy: `File space management strategy: H5F_FSPACE_STRATEGY_FSM_AGGR`. This is the default option. The best choice for cloud-optimized access is `H5F_FSPACE_STRATEGY_PAGE`. Learn more about the options in the HDF docs: [File Space Management (HDF Group)](https://support.hdfgroup.org/documentation/hdf5-docs/advanced_topics/FileSpaceManagement.html).
+
+## How to change the file space management strategy
+
+You can use the `h5repack` to reorganize the metadata [@jelenak2023]. When repacking to use the `PAGE` file space management strategy, you will also need to specify a page size that will indicate the block size for metadata storage. This should be at least as big as the `File metadata` value returned from `h5stat -S`.
+
+``` bash
+$ h5repack -S PAGE -G 4000000 infile.h5 outfile.h5
+```
+
+The HDF5 library needs to be configured to use the page aggregated files. When using the HDF5 library you can set  [H5Pset_page_buffer_size](https://hdfgroup.github.io/hdf5/develop/group___f_a_p_l.html#title89) and for [h5py File objects](https://docs.h5py.org/en/stable/high/file.html) you can set `page_buf_size` when instantiating the File object.
```

Comment:
I'm confused, is this about writing or about reading - did I miss an example of passing this higher up on the page when it's discussed also? Might just need a few more words clarifying.

Reply:
Thanks for identifying that this is confusing - I made changes in https://github.com/cloudnativegeo/cloud-optimized-geospatial-formats-guide/pull/126/commits/c0a9d282383862e836c433448eeb01fb4743b99f which hopefully clears things up.